app/src/main/java/com/example/movieapp/SingleFragmentActivity.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.movieapp
+
+import android.os.Bundle
+import androidx.annotation.LayoutRes
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import com.example.movieapp.viewModel.MainViewModel
+
+abstract class SingleFragmentActivity : AppCompatActivity() {
+    lateinit var viewModel: MainViewModel
```

Comment:
Why do you need to have access to MainViewModel in the activity?

Reply:
No need, job done